### DISCUSSION TITLE
Is it possible to run libvips for microcontroller based system?

### DISCUSSION THREAD

**Discussion by shrikantvaishnav** [libvips/libvips#3820] [General]
Hi all,
I am new to "libvips" and want to know that:is it possible to run **libvips** for **micro-controller** based system as i am working on **"Graphic LCD"** and i want to perform some graphic related operations.
Thanks
Shrikant Vaishnav

**Comment by jcupitt** [comment]
Hello, lib vips needs glib, and glib mostly needs linux-like systems. You'd need quite a chunky microcontroller.

**Reply by shrikantvaishnav** [reply]
sir that means its not possible to run libvips in microcontroller based system,any alternative available as i am new sir?

**Reply by jcupitt** [reply]
Sorry :( I don't know.

**Reply by kmartinez** [reply]
as John says the microcontroller would have to be running a linux kernel - so its not an easy task at the moment as most microcontrollers run a tiny operating system or none. if you are drawing to a small LCD and don't need much image processing you can use something like micropython/circuitpython or arduino.
